### PR TITLE
MFA Config with TOTP Provider

### PR DESCRIPTION
**Pull request by pragatimodi**

### CODE REVIEW

File: auth/config.go [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+	constMfaObject                = "MultiFactorConfig"
+	constProviderConfigObject     = "ProviderConfig"
+	constTotpProviderConfigObject = "TotpProviderConfig"
+	constState                    = "state"
+	constEnabledProviders         = "enabledProviders"
+	constFactorIds                = "factorIds"
+)
+
+var validAuthFactors = map[string]bool{
+	"PHONE_SMS": true,
```
Comment (renkelvin) [inline auth/config.go:24]: Should this include SMS or TOTP?

Comment (pragatimodi) [reply auth/config.go:24]: SMS only, we've switch to `providerConfigs` for factors other than SMS.

**Review by renkelvin — APPROVED** [review]
LGTM, thanks!

File: auth/config.go [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+func validateConfigKeys(inputReq *map[string]interface{}, validKeys map[string]bool, configName string) error {
+	for key := range *inputReq {
+		if !validKeys[key] {
+			return fmt.Errorf(`"%s" is not a valid %s parameter`, key, configName)
+		}
+	}
+	return nil
+}
+
+func stringKeys(m *map[string]bool) string {
```
[the remark this answers is not in the document]
Comment (pragatimodi) [reply auth/config.go:63]: I meant string as a verb but I get that it could be confusing.
Thanks for pointing this out!

**Pull request closed without merging**